### ISSUE TITLE
Honor __HIP_PLATFORM_AMD__ in all ROCM components

### ISSUE THREAD

**Issue by ye-luo** [RadeonOpenCompute/ROCm#1566]
Hit the issue of `hipsparse.h` not picking up `__HIP_PLATFORM_AMD__`
```
/opt/rocm-4.3.0/include$ grep -R _PLATFORM_HCC__ |grep  -v __HIP_PLATFORM_AMD__
hipsparse.h:#if defined(__HIP_PLATFORM_HCC__)
rocrand/rocrand_common.h:#if __HIP_DEVICE_COMPILE__ && (defined(__HIP_PLATFORM_HCC__) || (defined(__HIP_PLATFORM_NVCC__) && (__CUDA_ARCH__ >= 530)))
rocrand/rocrand_common.h:  #if defined(__HIP_PLATFORM_HCC__) && defined(__HIP_DEVICE_COMPILE__) \
rocrand/rocrand_philox4x32_10.h:        #if defined(__HIP_PLATFORM_HCC__)
hip/hip_common.h:#ifndef __HIP_PLATFORM_HCC__
hip/hip_common.h:#define __HIP_PLATFORM_HCC__
hip/hip_runtime.h:#ifdef __HIP_PLATFORM_HCC__
hip/hip_runtime.h:#ifdef __HIP_PLATFORM_HCC__
hiprand/hiprand_kernel.h:#ifdef __HIP_PLATFORM_HCC__
hiprand/hiprand.h:#if defined(__HIP_PLATFORM_HCC__)
hiprand/hiprand_mtgp32_host.h:#if defined(__HIP_PLATFORM_HCC__) || HIPRAND_DOXYGEN
hiprand/hiprand_mtgp32_host.h:#endif // __HIP_PLATFORM_HCC__
```
Please check all the components

**Comment by ROCmSupport** [comment]
Thanks @ye-luo for reaching out.
I will check this for you and get back with an update.
Thank you.

**Comment by doctorcolinsmith** [comment]
Thanks for the report @ye-luo.  Can you please add which GPU and OS you are using?

**Comment by ye-luo** [comment]
Ubuntu 20.4. Radeon VII. Why GPU matters, I don't need a GPU to compile code.

**Comment by ROCmSupport** [comment]
Hi @ye-luo 
Good news. This issue is tracked and fixed as part of 4.5, can you please verify and update asap.
Thank you.

**Comment by ye-luo** [comment]
4.5 is good
